src/main/java/com/thebois/models/inventory/IInventory.java

```diff
@@ -77,4 +91,13 @@
      */
     int numberOf(ItemType itemType);
 
+    /**
+     * Calculates and returns a list of the differing IItems between two inventories.
+     *
+     * @param inventory The inventory to check against.
+     *
+     * @return A list of differing items.
+     */
+    ArrayList<ItemType> calculateDifference(IInventory inventory);
```

Comment:
I can use this and an empty inventory to extract all of the item types in an inventory. That's a behavior that isn't really intended, but as they say, if something is public then someone somewhere will depend on it. 

Is there no other way to calculate the differing items? Could you not just use the counts of the inventories to calculate how far a structure has been built?

Reply:
I need a way to compare two list. I don't see how that could be done without being able to check what's inside? But maybe there is a way around it?

Reply:
Sure. I just read what you wrote about it further down, I could take a look if I can apply that idea